kernel/kernel-api/src/test/scala/io/delta/kernel/internal/stats/FileSizeHistogramSuite.scala

```diff
@@ -269,6 +269,100 @@ class FileSizeHistogramSuite extends AnyFunSuite {
     assert(histogram === reconstructedHistogram.get())
   }
 
+  test("plus should combine histograms correctly") {
+    // Create two histograms
+    val histogram1 = FileSizeHistogram.createDefaultHistogram()
+    val histogram2 = FileSizeHistogram.createDefaultHistogram()
+
+    // Add data to first histogram
+    histogram1.insert(1 * MB)
+    histogram1.insert(1 * MB)
+    histogram1.insert(10 * MB)
+
+    // Add data to second histogram
+    histogram2.insert(2 * MB)
+    histogram2.insert(10 * MB)
+    histogram2.insert(10 * MB)
+
+    // Combine histograms
+    val combined = histogram1.plus(histogram2)
+
+    // Check results
+    val mbBinIndex1 = getBinIndexForTesting(combined.getSortedBinBoundaries, 1 * MB)
+    val mbBinIndex2 = getBinIndexForTesting(combined.getSortedBinBoundaries, 2 * MB)
+    val mbBinIndex10 = getBinIndexForTesting(combined.getSortedBinBoundaries, 10 * MB)
+
+    assert(combined.getFileCounts(mbBinIndex1) == 2)
+    assert(combined.getTotalBytes(mbBinIndex1) == 2 * MB)
+
+    assert(combined.getFileCounts(mbBinIndex2) == 1)
+    assert(combined.getTotalBytes(mbBinIndex2) == 2 * MB)
+
+    assert(combined.getFileCounts(mbBinIndex10) == 3)
+    assert(combined.getTotalBytes(mbBinIndex10) == 30 * MB)
+
+    // Test error case - different bin boundaries
+    val customBoundaries = Array(0L, 1L * KB, 1L * MB)
+    val customCounts = new Array[Long](customBoundaries.length)
+    val customBytes = new Array[Long](customBoundaries.length)
+    val customHistogram = new FileSizeHistogram(customBoundaries, customCounts, customBytes)
+
+    intercept[IllegalArgumentException] {
+      histogram1.plus(customHistogram)
+    }
+  }
+
+  test("minus should subtract histograms correctly") {
```

Comment:
similar idea, try add two and then subtract that and are equal? super easy to test combos of these operations

Reply:
Added a case for add then subtract returns the original histogram